sigridci/sigridci/sigrid_api_client.py

```diff
@@ -104,9 +104,11 @@ def handle401or403(e: urllib.error.HTTPError, msg: str):
 
     @staticmethod
     def printResponse(e: urllib.error.HTTPError):
-        UploadLog.log(f"Response headers:\n{str(e.headers)}")
-        if e.fp is not None:
-            UploadLog.log(f"Response body:\n{e.fp.read().decode()}")
+        UploadLog.log(f"Response headers:\n{dict(e.headers)}")
+
+        body = e.fp.read().decode("utf8") if e.fp else ""
+        if body:
```

Comment:
Suppose `e.fp` is falsy, then `body` is the empty string (previous line). In all other cases, `body` is the result of `decode("utf8")`. Doesn't this imply that body is always true(-like)?

Reply:
An empty string is falsy, so if the response body is empty you would get the "no body" message.